# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webgpu/Device.cpp
# Commit: a5be865d06b1
# Full Hash: a5be865d06b167e20cff16ad33e4f076bbe41757
# Author: Erich Gubler <erichdongubler@gmail.com>
# Date: 2025-07-17 09:30:54
# Regressor Bug: 1976960
# File Overlap Count: 1
# Description:
#   Bug 1976960 - feat(webgpu): allow `GPUTexture`s where `GPUTextureView`s are allowed r=webgpu-reviewers,webidl,smaug,nical
#   
#   Changed in spec. upstream by [gpuweb#5228](https://github.com/gpuweb/gpuweb/pull/5228), tested by CTS in [gpuweb/cts#4407](https://github.com/gpuweb/cts/pull/4407) (which is what caused the `FAIL`s removed here to be introduced with bug 1976874).
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D257054
# ==============================================================================

diff -r e98afaa9f97b -r a5be865d06b1 dom/webgpu/Device.cpp
--- a/dom/webgpu/Device.cpp	Thu Jul 17 04:08:37 2025 +0000
+++ b/dom/webgpu/Device.cpp	Thu Jul 17 04:08:38 2025 +0000
@@ -465,6 +465,11 @@
       e.buffer = bufBinding.mBuffer->mId;
       e.offset = bufBinding.mOffset;
       e.size = bufBinding.mSize.WasPassed() ? bufBinding.mSize.Value() : 0;
+    } else if (entry.mResource.IsGPUTexture()) {
+      auto texture = entry.mResource.GetAsGPUTexture();
+      const dom::GPUTextureViewDescriptor defaultDesc{};
+      RefPtr<TextureView> texture_view = texture->CreateView(defaultDesc);
+      setTextureViewBinding(*texture_view);
     } else if (entry.mResource.IsGPUTextureView()) {
       auto texture_view = entry.mResource.GetAsGPUTextureView();
       setTextureViewBinding(texture_view);